### DISCUSSION TITLE
Question: Where to put OpenAPIs folder ?

### DISCUSSION THREAD

**Discussion by Inexad** [jasontaylordev/CleanArchitecture#484] [Q&A]
Where should i put OpenAPIs folder ?

Currently i put it in Application folder, is this correct ? 

![image](https://user-images.githubusercontent.com/8613836/135979389-31858fd1-c441-4bd8-b8c0-0f39c0fec345.png)

**Comment by bommen** [comment]
I would expect this to be in the WebUI project because it's part of the presentation layer. The Application project should be independent of the technology used for presentation (REST, GraphQL, MVC, etc).

**Comment by Inexad** [comment]
@bommen What if i need to use the data from this in Services within my Infrastructure? It's not OK to reference within Infrastructure to WebUI?

**Comment by bommen** [comment]
@Inexad I'm curious what the specific case is where you would reference your OpenAPI doc (swagger.json) in your Application or Infrastructure layer. Could you give an example of why you would need this?

**Comment by Inexad** [comment]
I use this to generate service class library. This was added automatically when using "Connected Services" in Visual Studio 2019. I guess it downloads the open api spec file and then generates code from it.

**Comment by fretje** [comment]
@Inexad If I understand correctly, the service class library you generated using "connected services" in Visual Studio 2019 is a third party service you need to call? 
If so that means it is an external service and it should go into infrastructure. If you need to call it from your application layer, you should create an interface for it in the application project and use that there. The interface is then implemented in the infrastructure project. Similar like you would only use an IRepository interface in the application layer for database access, but the actual repositories are implemented in the infrastructure layer.

**Comment by bommen** [comment]
In that case you appear to be consuming the API in the WebUI project from within the Infrastructure project. I would place the folder in the Infrastructure project.

**Comment by Inexad** [comment]
Thanks ! I put it in Infrastructure project. @fretje What do you think about "Settings" classes? Thinking about appsettings injected into controllers using IOptionsMonitor.
